fineract-investor/src/main/java/org/apache/fineract/investor/api/ExternalAssetOwnerLoanProductAttributesApiResource.java

```diff
@@ -0,0 +1,140 @@
+import org.apache.fineract.infrastructure.core.service.Page;
+import org.apache.fineract.infrastructure.security.service.PlatformUserRightsContext;
+import org.apache.fineract.investor.config.InvestorModuleIsEnabledCondition;
+import org.apache.fineract.investor.data.ExternalTransferLoanProductAttributesData;
+import org.apache.fineract.investor.service.ExternalAssetOwnerLoanProductAttributesReadService;
+import org.springframework.context.annotation.Conditional;
+import org.springframework.stereotype.Component;
+
+@Path("/v1/external-asset-owners/loan-product")
+@Component
+@Tag(name = "External Asset Owner Loan Product Attributes", description = "External Asset Owner Loan Product Attributes")
+@RequiredArgsConstructor
+@Conditional(InvestorModuleIsEnabledCondition.class)
+public class ExternalAssetOwnerLoanProductAttributesApiResource {
+
+    private final PlatformUserRightsContext platformUserRightsContext;
+    private final PortfolioCommandSourceWritePlatformService commandsSourceWritePlatformService;
+    private final ExternalAssetOwnerLoanProductAttributesReadService externalAssetOwnerLoanProductAttributesReadService;
+    private final ApiRequestParameterHelper apiRequestParameterHelper;
+    private final DefaultToApiJsonSerializer<ExternalTransferLoanProductAttributesData> toApiJsonSerializer;
+
+    @POST
+    @Path("/{loanProductId}/attributes")
+    @Consumes({ MediaType.APPLICATION_JSON })
+    @Produces({ MediaType.APPLICATION_JSON })
+    @RequestBody(required = true, content = @Content(schema = @Schema(implementation = ExternalAssetOwnerLoanProductAttributesApiResourceSwagger.PostExternalAssetOwnerLoanProductAttributeRequest.class)))
+    @ApiResponses({
+            @ApiResponse(responseCode = "200", description = "OK", content = @Content(schema = @Schema(implementation = ExternalAssetOwnerLoanProductAttributesApiResourceSwagger.PostExternalAssetOwnerLoanProductAttributeResponse.class))),
+            @ApiResponse(responseCode = "400", description = "Invalid Request"),
+            @ApiResponse(responseCode = "409", description = "Resource Already Exists"),
+            @ApiResponse(responseCode = "404", description = "Not Found"),
+            @ApiResponse(responseCode = "500", description = "Internal Server Error") })
+    public String postExternalAssetOwnerLoanProductAttribute(
+            @PathParam("loanProductId") @Parameter(description = "loanProductId") final Long loanProductId,
+            @Parameter(hidden = true) final String apiRequestBodyAsJson) {
+        platformUserRightsContext.isAuthenticated();
+        final CommandWrapperBuilder builder = new CommandWrapperBuilder().withJson(apiRequestBodyAsJson);
+        CommandWrapper request = builder.createExternalAssetOwnerLoanProductAttribute(loanProductId).build();
+
+        CommandProcessingResult result = this.commandsSourceWritePlatformService.logCommandSource(request);
+        return toApiJsonSerializer.serialize(result);
```

Comment:
We can skip this part and return the `CommandProcessingResult` directly. This way we dont even need the `PostExternalAssetOwnerLoanProductAttributeRespons` either.

Reply:
done.